pkg/pod/inspector.go

```diff
@@ -27,29 +29,29 @@ func NewInspector(pods podGetter, namespaces []string, handlers ...Handler) Insp
 
 func (i Inspector) Inspect() error {
 	handlerMap := i.createHandlersByLabelMap()
-	log.Printf("Found %d handlers\n", len(i.handlers))
+	i.logger.Info("Handlers loaded", "count", slog.IntValue(len(i.handlers)))
 
 	pods, err := i.pods.Get(i.namespaces)
 	if err != nil {
 		return err
 	}
 
-	log.Printf("Inspecting %d pods\n", len(pods))
+	i.logger.Info("Inspecting pods", "count", slog.IntValue(len(pods)))
 	for _, pod := range pods {
-		log.Printf("Target: %s", pod.Name)
+		i.logger.Info("Inspecting pod", "pod", pod.Name)
 		for label := range pod.Labels {
 			handlers, exists := handlerMap[label]
 			if exists && strings.HasPrefix(label, "im-") {
 				for _, h := range handlers {
 					err := h.Handle(pod)
 					if err != nil {
-						log.Println(err)
+						i.logger.Info(err.Error())
 					}
 				}
 			}
 		}
 	}
-	log.Println("Inspection ended!")
+	i.logger.Info("Inspection done!")
```

Comment:
```suggestion
	i.logger.Info("Inspection done")
```

You could keep a symmetry between the start/end of inspection in the log message. This might make it easier to find in loki. Maybe `Start inspection`/`End inspection`.